README.md

```diff
@@ -56,83 +56,140 @@ bash ./benches/generate.sh
+keypairs and the master secret key holds the secret key of all these keypairs.
+
+When encrypting for a given encryption policy, this policy is converted into a
+set of partitions. Then, one key encapsulation is generated per partition using
+the corresponding public sub-key in the master public key.
+
+Similarly, when generating a user secret key for a given user policy, this
+policy is converted into the set of corresponding partitions and the user
+receives the secret sub-key associated to each partitions.
+
+**Example**: the following policy is composed of two axes. The `Security` axis
+composed of three attributes and the `Country` axis composed of 4 attributes.
+```txt
+Polixy: {
+	Security: {	// <- first axis
+		None,
+		Medium,
+		High
+	},
+	Country: {	// <- second axis
+		France,
+		Germany,
+		UK,
+		Spain
+	}
+}
+```
+The encryption policy `Security::Medium && ( Country::France ||
+Country::Spain)` would be converted into two partitions. The encryption policy
+`Security::High` would be expanded into `Security::High && (Country::France ||
+... || Country::Spain)` then converted into 4 partitions.
 
 ### Serialization
 
 The size of the serialized keys and encapsulation is given by the following formulas:
 
 - master secret key:
-
-```c
-3 * PRIVATE_KEY_LENGTH + LEB128_sizeof(partitions.len()) \
-    + sum(LEB128_sizeof(sizeof(partition)) + sizeof(partition)
-  + PRIVATE_KEY_LENGTH + 1 [+ INDCPA_KYBER_PRIVATE_KEY_LENGTH])
-```
+$$3 \cdot L_{sk} + \texttt{LEB128sizeof}(n_{p}) + \sum\limits_{p~\in~partitions} \left( \texttt{LEB128sizeof}(\texttt{sizeof}(p)) + \texttt{sizeof}(p) + 1 + L_{sk} + \delta_{p,~h} \cdot L_{sk}^{pq}\right)$$
 
 - public key:
 
-```c
-2 * PUBLIC_KEY_LENGTH + LEB128_sizeof(partitions.len()) \
-    + sum(LEB128_sizeof(sizeof(partition)) + sizeof(partition)
-    + PUBLIC_KEY_LENGTH + 1 [+ INDCPA_KYBER_PUBLIC_KEY_LENGTH])
-```
+$$3 \cdot L_{pk} + \texttt{LEB128sizeof}(n_{p}) + \sum\limits_{p~\in~partitions} \left( \texttt{LEB128sizeof}(\texttt{sizeof}(p)) + \texttt{sizeof}(p) + 1 + L_{pk} + \delta_{p,~h} \cdot L_{pk}^{pq}\right)$$
 
 - user secret key:
 
-```c
-2 * PRIVATE_KEY_LENGTH + LEB128_sizeof(partitions.len()) \
-    + partition.len() * (PRIVATE_KEY_LENGTH + 1 [+ INDCPA_KYBER_PRIVATE_KEY_LENGTH])
-```
+$$2 \cdot L_{sk} + \texttt{LEB128sizeof}(n_{p}) + \sum\limits_{p~\in~partitions} \left( 1 + L_{sk} + \delta_{p,~h} \cdot L_{sk}^{pq}\right)$$
 
 - encapsulation:
 
-```c
-2 * PUBLIC_KEY_LENGTH + TAG_LENGTH + LEB128_sizeof(partitions.len())
- + partition.len() * [INDCPA_KYBER_CIPHERTEXT_LENGTH | PUBLIC_KEY_LENGTH]
-```
+$$2 \cdot L_{pk} + T + \texttt{LEB128sizeof}(n_{p}) + \sum\limits_{p~\in~partitions} \left(1 + \delta_{p,~c} \cdot L_{pk} + \delta_{p,~h} \cdot L_c^{pq}\right)$$
 
-- encrypted header (see below):
+- encrypted header:
 
-```c
-sizeof(encapsulation) + DEM_ENCRYPTION_OVERHEAD + sizeof(plaintext)
-```
+$$\texttt{sizeof}(encapsulation) + C_{encryption~overhead} + \texttt{sizeof}(plaintext)$$
```

Comment:
Could we mention the metadata in the `encrypted header`? 
Because in the tables of section `Secret Key encapsulation`, we have a `+1` related to the metadata-length serialization (when metadata is null).
Maybe we could remove 1 in the values of  `Encapsulation size` column of those tables but still mention the metadata in the `encrypted header` to remove any doubt.

Reply:
Indeed, this is a mistake. I added the LEB128 length of the encrypted metadata in the encrypted header serialization.
And yes you are right, I will remove the `+1` in the tables.

Reply:
done in da5f781